### PR TITLE
add function admission_control()

### PR DESCRIPTION
**Pull request by alicelee0606**
Why?
======
function admissionControl() is for step 8
How?
======
by using fixed point arithmetic
cc @raywang945 @maple1216

### CODE REVIEW

File: MP2/mp2.c [hunk 1]
```diff
@@ -60,6 +60,27 @@ proc_list *get_highest_priority_ready_task(void)
     return result;
 }
 
+static bool admission_control(unsigned int computation, unsigned int period)
```
Comment (raywang945) [inline MP2/mp2.c]: Does the returning type bool work?

File: MP2/mp2.c [hunk 2]
```diff
@@ -60,6 +60,27 @@ proc_list *get_highest_priority_ready_task(void)
...
+    mutex_lock_interruptible(&mp2_mutex);
+    list_for_each_entry(tmp, &mp2_proc_list, list) {
+        sum_C += tmp->computation;
+        sum_P += tmp->period;
+    }
+    mutex_unlock(&mp2_mutex);
+    sum_C += computation;
+    sum_P += period;
+    if (sum_C*multiplier <= util_bound*sum_P)
+       return true;
```
Comment (raywang945) [inline MP2/mp2.c]: @alicelee0606 add brackets for the if else statements, thank you!

File: MP2/mp2.c [hunk 3]
```diff
@@ -103,10 +124,16 @@ void mp2_register_processs(char *buf)
...
-    list_add(&(tmp->list), &mp2_proc_list);
-    mutex_unlock(&mp2_mutex);
+    if (admission_control(tmp->computation,tmp->period))
+    {
+        // add tmp to mp2_proc_list
+        printk(KERN_INFO "peace register!\n");
+        mutex_lock_interruptible(&mp2_mutex);
+        list_add(&(tmp->list), &mp2_proc_list);
+        mutex_unlock(&mp2_mutex);
+    }else{
```
Comment (raywang945) [inline MP2/mp2.c]: @alicelee0606 whitespace before the else statement, thank you!

File: MP2/mp2.c [hunk 4]
```diff
@@ -60,6 +60,27 @@ proc_list *get_highest_priority_ready_task(void)
     return result;
 }
 
+static bool admission_control(unsigned int computation, unsigned int period)
+{
+    unsigned int sum_C = 0;
+    unsigned int sum_P = 0;
+    unsigned int util_bound = 693;
+    unsigned int multiplier = 1000;
+    proc_list *tmp;
```
Comment (raywang945) [inline MP2/mp2.c]: @alicelee0606 add blank line between line 69 and 70

File: MP2/mp2.c (hunk 1 above)
Comment (alicelee0606) [inline MP2/mp2.c]: Yes. I think it works~ I had compiled and run it.

File: MP2/mp2.c [hunk 5]
```diff
@@ -60,6 +60,27 @@ proc_list *get_highest_priority_ready_task(void)
...
+    mutex_lock_interruptible(&mp2_mutex);
+    list_for_each_entry(tmp, &mp2_proc_list, list) {
+        sum_C += tmp->computation;
+        sum_P += tmp->period;
+    }
+    mutex_unlock(&mp2_mutex);
```
Comment (raywang945) [inline MP2/mp2.c]: @alicelee0606 add blank line between line 75 and 76

File: MP2/mp2.c (hunk 2 above)
Comment (alicelee0606) [inline MP2/mp2.c]: Done.

File: MP2/mp2.c (hunk 3 above)
Comment (alicelee0606) [inline MP2/mp2.c]: Done.

File: MP2/mp2.c [hunk 6]
```diff
@@ -103,10 +124,16 @@ void mp2_register_processs(char *buf)
     // set tmp->state
     tmp->state = SLEEPING;
 
-    // add tmp to mp2_proc_list
-    mutex_lock_interruptible(&mp2_mutex);
-    list_add(&(tmp->list), &mp2_proc_list);
-    mutex_unlock(&mp2_mutex);
+    if (admission_control(tmp->computation,tmp->period))
+    {
```
Comment (raywang945) [inline MP2/mp2.c]: @alicelee0606 put this bracket back to line 127, thank you!

File: MP2/mp2.c (hunk 4 above)
Comment (alicelee0606) [inline MP2/mp2.c]: Done.

File: MP2/mp2.c (hunk 5 above)
Comment (alicelee0606) [inline MP2/mp2.c]: Done.

File: MP2/mp2.c (hunk 6 above)
Comment (alicelee0606) [inline MP2/mp2.c]: Done.

**Pull request merged**